src/client/app/components/HeaderButtonsComponent.tsx

```diff
@@ -24,33 +24,14 @@ export default class HeaderButtonsComponent extends React.Component<HeaderButton
 	}
 
 	public render() {
-		let showOptions = false;
-		let renderLoginButton = false;
-		let renderHomeButton = true;
-		let renderAdminButton = false;
-		let renderGroupsButton = true;
+		const showOptions = false;
+		const renderLoginButton = !hasToken();
+		const renderHomeButton = getPage() !== '';
+		const renderAdminButton = hasToken() && getPage() !== 'admin';
+		const renderGroupsButton = hasToken() && getPage() !== 'groups';
```

Comment:
I think we don't need the hasToken() check here as the current development branch allows rendering the button without logging in.

Reply:
Like this?

```suggestion
		const renderGroupsButton = getPage() !== 'groups';
```

Reply:
Yes